scripts/gen-doc.sh

```diff
@@ -28,39 +28,42 @@ DOCS_REPO="$(dirname "$SOF_REPO")/sof-docs"
 PUBLISH_REPO="$(dirname "$SOF_REPO")/thesofproject.github.io"
 
 # parse arguments
-DO_BUILD=no
-DO_CLEAN=no
-DO_PUBLISH=no
-DO_CLONE=no
-
-for args in $@
-do
-	if [[ "$args" == "-b" ]]
-	then
-		DO_BUILD=yes
-	elif [[ "$args" == "-c" ]]
-	then
-		DO_CLEAN=yes
-	elif [[ "$args" == "-p" ]]
-	then
-		DO_PUBLISH=yes
-	elif [[ "$args" == "-r" ]]
-	then
-		DO_CLONE=yes
-	fi
+DO_BUILD=false
+DO_CLEAN=false
+DO_PUBLISH=false
+FETCH_MISSING=false
+
+while getopts "bcpr" OPTION; do
+        case "$OPTION" in
+        b)
+                DO_BUILD=true
+                ;;
+        c)
+                DO_CLEAN=true
+                ;;
+        p)
+                DO_PUBLISH=true
+                ;;
+        r)
+                FETCH_MISSING=true
+                ;;
+        *)
+                print_usage
+                exit 1
+                ;;
+        esac
 done
+shift "$(($OPTIND - 1))"
 
-if [ "x$DO_BUILD" == "xno" ] && \
-	[ "x$DO_CLEAN" == "xno" ] && \
-	[ "x$DO_PUBLISH" == "xno" ]
+if ! { [ "$DO_BUILD" ] || [ "$DO_CLEAN" ] || [ "$DO_PUBLISH" ]; }
```

Comment:
`[ "false" ]` is true because it's a non-empty string, see `help test`. Did you test this part of the code? I don't think it will ever print the usage.

Try these in an interactive shell and compare: `false; echo $?` with `[ "false" ] ; echo $?`

What you want is to **run** the command `true` or `false` like this:
```bash
$DO_BUILD || $DO_CLEAN || $DO_PUBLISH || {
   print_usage; exit 1
}
```
While you should always double-quote, it's overkill for a local variable that you know for sure can only have two, whitespace-free values completely under control. Up to you.

Reply:
This is a good explanation, thank you! And yes, this block of code wasn't getting executed ever.